Dockerfile

```diff
@@ -1,87 +1,65 @@
 # syntax=docker/dockerfile:1
 ARG PYTHON_VERSION=3.10
-FROM python:$PYTHON_VERSION-slim AS base
+FROM ghcr.io/astral-sh/uv:python$PYTHON_VERSION-bookworm AS dev
 
-# Remove docker-clean so we can keep the apt cache in Docker build cache.
-RUN rm /etc/apt/apt.conf.d/docker-clean
-
-# Configure Python to print tracebacks on crash [1], and to not buffer stdout and stderr [2].
-# [1] https://docs.python.org/3/using/cmdline.html#envvar-PYTHONFAULTHANDLER
-# [2] https://docs.python.org/3/using/cmdline.html#envvar-PYTHONUNBUFFERED
-ENV PYTHONFAULTHANDLER 1
-ENV PYTHONUNBUFFERED 1
-
-# Create a non-root user and switch to it [1].
-# [1] https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user
-ARG UID=1000
-ARG GID=$UID
-RUN groupadd --gid $GID user && \
-    useradd --create-home --gid $GID --uid $UID user --no-log-init && \
-    chown user /opt/
-USER user
-
-# Create and activate a virtual environment.
-ENV VIRTUAL_ENV /opt/raglite-env
-ENV PATH $VIRTUAL_ENV/bin:$PATH
-RUN python -m venv $VIRTUAL_ENV
-
-# Set the working directory.
-WORKDIR /workspaces/raglite/
+# Create and activate a virtual environment [1].
+# [1] https://docs.astral.sh/uv/concepts/projects/config/#project-environment-path
+ENV VIRTUAL_ENV=/opt/venv
+ENV PATH=$VIRTUAL_ENV/bin:$PATH
+ENV UV_PROJECT_ENVIRONMENT=$VIRTUAL_ENV
 
+# Tell Git that the workspace is safe to avoid 'detected dubious ownership in repository' warnings.
+RUN git config --system --add safe.directory '*'
 
+# Configure the user's shell.
+RUN echo 'HISTFILE=~/.history/.bash_history' >> ~/.bashrc && \
+    echo 'bind "\"\e[A\": history-search-backward"' >> ~/.bashrc && \
+    echo 'bind "\"\e[B\": history-search-forward"' >> ~/.bashrc && \
+    mkdir ~/.history/
 
-FROM base AS poetry
+FROM python:$PYTHON_VERSION-slim AS app
```

Comment:
Normally with the `package` option we should not have an `app` stage.

Reply:
Oh indeed, my bad. Thanks for catching this one!